src/main/java/de/tum/in/www1/artemis/service/StaticCodeAnalysisService.java

```diff
@@ -142,6 +142,32 @@ public Set<StaticCodeAnalysisCategory> resetCategories(ProgrammingExercise exerc
         return categories;
     }
 
+    /**
+     * This method allows users to reuse an already existing SCA configuration by copying it into another exercise.
+     * The previous configuration of the targeted exercise will get removed.
+     * @param sourceExercise The exercise to take the existing configuration from
+     * @param targetExercise The exercise into which the configuration gets copied in
+     * @return the new SCA configuration of the targetExercise
+     */
+    public Set<StaticCodeAnalysisCategory> importCategoriesFromExercise(ProgrammingExercise sourceExercise, ProgrammingExercise targetExercise) {
+        var sourceCategories = findByExerciseId(sourceExercise.getId());
+        var oldCategories = findByExerciseId(targetExercise.getId());
+
+        var newCategories = sourceCategories.stream().map(category -> {
+            var copy = category.copy();
+            copy.setProgrammingExercise(targetExercise);
+            return copy;
+        }).collect(Collectors.toSet());
```

Comment:
I'd keep it like that.

From the peek-JavaDoc:
> API Note:
    This method exists mainly to support debugging, where you want to see the elements as they flow past a certain point in a pipeline

Since we change the object, we should use map() here.